superpoint/evaluations/detector_evaluation.py

```diff
@@ -0,0 +1,222 @@
+    paths = get_paths(exper_name)
+    error = []
+    for path in paths:
+        error.append(loc_error_per_image(np.load(path)))
+    return np.mean(np.concatenate(error))
+
+
+def compute_repeatability(exper_name, prob_thresh=0.5, distance_thresh=3):
+    """
+    Compute the repeatability. The experiment must contain in its output the prediction
+    on 2 images, an original image and a warped version of it, plus the homography
+    linking the 2 images.
+    """
+    def warp_keypoints(keypoints, H):
+        warped_col0 = np.add(np.sum(np.multiply(keypoints, H[0, :2]), axis=1), H[0, 2])
+        warped_col1 = np.add(np.sum(np.multiply(keypoints, H[1, :2]), axis=1), H[1, 2])
+        warped_col2 = np.add(np.sum(np.multiply(keypoints, H[2, :2]), axis=1), H[2, 2])
+        warped_col0 = np.divide(warped_col0, warped_col2)
+        warped_col1 = np.divide(warped_col1, warped_col2)
+        new_keypoints = np.concatenate([warped_col0[:, None], warped_col1[:, None]],
+                                       axis=1)
+        return new_keypoints
+
+    def filter_keypoints(points, shape):
+        """ Keep only the points whose coordinates are
+        inside the dimensions of shape. """
+        mask = (points[:, 0] >= 0) & (points[:, 0] < shape[0]) &\
+               (points[:, 1] >= 0) & (points[:, 1] < shape[1])
+        return points[mask, :]
+
+    def keep_true_keypoints(points, H, shape):
+        """ Keep only the points whose warped coordinates by H
+        are still inside shape. """
+        warped_points = warp_keypoints(points[:, [1, 0]], H)
+        warped_points[:, [0, 1]] = warped_points[:, [1, 0]]
+        mask = (warped_points[:, 0] >= 0) & (warped_points[:, 0] < shape[0]) &\
+               (warped_points[:, 1] >= 0) & (warped_points[:, 1] < shape[1])
+        return points[mask, :]
+
+    paths = get_paths(exper_name)
+    repeatability = []
+    for path in paths:
+        data = np.load(path)
+        shape = data['warped_prob'].shape
+        H = data['homography']
+
+        # Filter out predictions
+        keypoints = np.where(data['prob'] > prob_thresh)
+        keypoints = np.stack([keypoints[0], keypoints[1]], axis=-1)
+        warped_keypoints = np.where(data['warped_prob'] > prob_thresh)
+        warped_keypoints = np.stack([warped_keypoints[0], warped_keypoints[1]], axis=-1)
+        warped_keypoints = keep_true_keypoints(warped_keypoints, np.linalg.inv(H),
+                                               data['prob'].shape)
+
+        # Warp the original keypoints with the true homography
+        true_warped_keypoints = warp_keypoints(keypoints[:, [1, 0]], H)
+        true_warped_keypoints[:, [0, 1]] = true_warped_keypoints[:, [1, 0]]
+        true_warped_keypoints = filter_keypoints(true_warped_keypoints, shape)
+
+        # Compute the repeatability
+        N1 = true_warped_keypoints.shape[0]
+        N2 = warped_keypoints.shape[0]
+        tile1 = np.tile(np.expand_dims(true_warped_keypoints, 1), (1, N2, 1))
```

Comment:
Tiling is unnecessary thanks to shape broadcasting (tensors still must have the rank though).